### PR TITLE
mrc-1548 Fetch baseline options from backend, and display in form

### PR DESCRIPTION
**Pull request by EmmaLRussell**
This PR:
- Fetches baseline options from the backend on initialise and saves to store
- Creates a copy of these options on each new region
- Displays the current region's options in the baseline form
- Saving and submitting options to be done in future tickets

### CODE REVIEW

File: src/app/static/src/app/actions.ts [hunk 1]
```diff
@@ -3,14 +3,23 @@ import {RootState} from "./store";
...
     FetchPrevalenceGraphData = "FetchPrevalenceGraphData",
-    FetchPrevalenceGraphConfig = "FetchPrevalenceGraphConfig"
+    FetchPrevalenceGraphConfig = "FetchPrevalenceGraphConfig",
+    FetchBaselineOptions = "FetchBaselineOptions"
 }
 
 export const actions: ActionTree<RootState, RootState> = {
 
+    async [RootAction.FetchBaselineOptions](context) {
+        await api(context)
```
Comment (EmmaLRussell) [inline src/app/static/src/app/actions.ts:17]: Unlike the other config data, I don't think we want to freeze this response because we'll be creating clones of it which must be mutable. I think if  you copy a frozen object you get another frozen object.

File: src/app/static/src/app/components/baseline.vue [hunk 2]
```diff
@@ -8,201 +8,26 @@
...
-                options: baselineOptions
-            }
+        computed: {
+            ...mapState(["currentProject"]),
+            options:  {
+                get() {
+                    return this.currentProject.currentRegion.baselineOptions
+                },
+                set(value: DynamicFormMeta) {
+                      //TODO: update by calling a mutation to set options in the current region
```
Comment (EmmaLRussell) [inline src/app/static/src/app/components/baseline.vue:28]: Code cov seems to be complaining about this, which will be implemented and tested in the next ticket..

File: src/app/static/src/app/actions.ts (hunk 1 above)
Comment (hillalex) [reply src/app/static/src/app/actions.ts:17]: that's fine, this isn't big data anyway :+1:

File: src/app/static/src/app/components/app.vue [hunk 3]
```diff
@@ -25,12 +25,24 @@
...
+        fetchBaselineOptions: () => void
+    }
+
+    export default Vue.extend<{}, Methods, {}, {}>({
+        store,
         components: {dropDown, BIconGraphUp},
-        computed: mapState(["currentProject"])
+        computed: mapState(["currentProject"]),
+        methods: {
+            ...mapActions({fetchBaselineOptions: "FetchBaselineOptions"})
```
Comment (hillalex) [inline src/app/static/src/app/components/app.vue:42]: this is where the action names enum comes in handy so you don't have to get the string right!

File: src/app/static/src/app/components/projectListPage.vue [hunk 4]
```diff
@@ -95,7 +99,7 @@
                     // so take this.newRegion as the only region
                     regionNames.push(this.newRegion)
                 }
-                const project = new Project(this.newProject, regionNames);
+                const project = new Project(this.newProject, regionNames, null, this.baselineOptions);
```
Comment (hillalex) [inline src/app/static/src/app/components/projectListPage.vue:102]: can we keep the nullable arg at the end of the method, so it can just be omitted?

File: src/app/static/src/app/models/project.ts [hunk 5]
```diff
@@ -1,12 +1,16 @@
...
 }
 
 export class Region {
-    constructor(name: string, parent: Project) {
-        this.name = name
-        this.url = `/projects/${parent.name}/regions/${name}`.replace(/\s/g, "-").toLowerCase()
+    constructor(name: string, parent: Project, baselineOptions: DynamicFormMeta) {
+        this.name = name;
+        this.url = `/projects/${parent.name}/regions/${name}`.replace(/\s/g, "-").toLowerCase();
+        this.baselineOptions = {...baselineOptions}; //Make our own copy
```
Comment (hillalex) [inline src/app/static/src/app/models/project.ts:13]: this won't be a deep copy, which I think will be a problem

Comment (EmmaLRussell) [reply src/app/static/src/app/models/project.ts:13]: Of course! :facepalm:

File: src/app/static/src/app/components/projectListPage.vue (hunk 4 above)
Comment (EmmaLRussell) [reply src/app/static/src/app/components/projectListPage.vue:102]: Yeah, I left it where it was to keep the two regions params together, but you're right it is neater to leave it to the end..

File: src/app/static/src/app/utils.ts [hunk 6]
```diff
@@ -16,6 +16,27 @@ export function isMINTResponseFailure(object: any): object is ResponseFailure {
         && object.errors.every((e: any) => isMINTError(e))
 }
 
+export function deepCopy(data: any): any {
+    let result = null;
+    if(!data) return result;
```
Comment (hillalex) [inline src/app/static/src/app/utils.ts:21]: this will return null if data is `0` or `false`

File: src/app/static/src/tests/utils.test.ts [hunk 7]
```diff
@@ -87,4 +87,25 @@ describe("utils", () => {
 
     });
 
+    it("deep copies an object", () => {
+
+        const data = {
+            nothing: null,
```
Comment (hillalex) [inline src/app/static/src/tests/utils.test.ts:93]: need some booleans, and a 0 in the test object, maybe an empty string as well
